### DISCUSSION TITLE
Javascript polling resets session inactivity duration

### DISCUSSION THREAD

**Discussion by bmoe3rd** [authelia/authelia#5231] [Q&A] [answered]
Hello fellow authelia users/devs,

currently i am developing a single page application (SPA) wich is secured by authelia.
As it is a SPA, it is wanted behaviour that xhr requests reset the inactivity duration of the session.

The Problem is, that i need to update the data on the Site continously, but i don't want to extend session life indefinitly.
Polling would be my prefered mechanism, but SSE might be possible as well.

Please enlighten me :)

**Comment by james-d-elliott** [comment, marked as answer]
Hi there. Any  request forwarded to Authelia which includes the appropriate `Cookie` session header will update the last activity value for the session and by extension will extend the inactivity  timeout.

**Comment by bmoe3rd** [comment]
Thank you, thats valuable information.

* So i could bypass authelia for the specific endpoint and use another security scheme.
Wich kind of defeats the purpose ...

* Another way would be to set the remember me duration to -1
wich would always destroy the session after expiration as inactivity would loose its meaning.

* SSE and Web Sockets
will most probably keep on working after the session gets destroyed
wich is undesireable (and untested)

* A mix between SSE and polling
Set remember_me_duration to -1
and poll at slightly longer intervals than inactivity
wich is no more secure than SSE alone,
but would allow for delayed redirection to login
(at the very least it is better than bypass without auth)

Better than polling an endpoint on the SPA would be polling authelia itself for authentication status.
Will polling the login page update the last activity value ?
